purchases/src/test/java/com/revenuecat/purchases/PurchasesTest.kt

```diff
@@ -1205,88 +1232,200 @@ class PurchasesTest {
             purchaseToken,
             observerMode = false,
             offeringIdentifier = null,
-            type = PurchaseType.INAPP
-        ) {
-            capturedLambda = lambda<PostReceiptDataErrorCallback>().captured
-            capturedLambda?.invokeWithFinishableError()
-        }
+            type = ProductType.INAPP,
+            answer = {
+                capturedLambda = lambda<PostReceiptDataErrorCallback>().captured
+                capturedLambda?.invokeWithFinishableError()
+            }
+        )
         mockPostReceiptError(
             skuSub,
             purchaseTokenSub,
             observerMode = false,
             offeringIdentifier = null,
-            type = PurchaseType.SUBS
-        ) {
-            capturedLambda = lambda<PostReceiptDataErrorCallback>().captured
-            capturedLambda?.invokeWithFinishableError()
+            type = ProductType.SUBS,
+            answer = {
+                capturedLambda = lambda<PostReceiptDataErrorCallback>().captured
+                capturedLambda?.invokeWithFinishableError()
+            }
+        )
+
+        val purchases = getMockedPurchaseList(sku, purchaseToken, ProductType.INAPP) +
+            getMockedPurchaseList(skuSub, purchaseTokenSub, ProductType.SUBS)
+        capturedPurchasesUpdatedListener.captured.onPurchasesUpdated(purchases)
+
+        purchases.forEach {
+            verify(exactly = 1) {
+                mockBillingAbstract.consumeAndSave(true, it)
+            }
         }
 
-        capturedPurchasesUpdatedListener.captured.onPurchasesUpdated(
-            getMockedPurchaseList(sku, purchaseToken, PurchaseType.INAPP) +
-                getMockedPurchaseList(skuSub, purchaseTokenSub, PurchaseType.SUBS)
+        assertThat(capturedLambda).isNotNull
+    }
+
+    @Test
+    fun `tries to consume restored purchases on 40x`() {
```

Comment:
these should be `4xx` and `5xx`, since you can get stuff like 429s. 
also maybe calling it `tries to consume restored purchases after 4xx error` is somewhat clearer? I didn't know what `40x` was referring to at first